atox/src/main/kotlin/ui/user_profile/UserProfileFragment.kt

```diff
@@ -155,23 +168,82 @@ class UserProfileFragment : BaseFragment<FragmentUserProfileBinding>(FragmentUse
     }
 
     private fun createQrCodeDialog(): AlertDialog {
+        val qrCode = ImageView(requireContext())
+        qrCode.setPadding(qrCodePadding.asPx(resources).px.toInt())
+        createQrCode(imageView = qrCode)
+
+        return AlertDialog.Builder(requireContext())
+            .setTitle(R.string.tox_id)
+            .setView(qrCode)
+            .setPositiveButton(getString(R.string.share)) { _, _ ->
+                vm.viewModelScope.launch {
+                    val qrImageUri = getQrForSharing("tox_id_qr_code")
+                    val shareIntent = Intent().apply {
+                        action = Intent.ACTION_SEND
+                        clipData = ClipData.newRawUri(null, qrImageUri)
+                        type = "image/png"
+                        putExtra(Intent.EXTRA_STREAM, qrImageUri)
+                        flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+                    }
+                    startActivity(Intent.createChooser(shareIntent, getString(R.string.tox_id_share)))
+                }
+            }
+            .setNegativeButton(getString(R.string.close)) { dialog, _ ->
+                dialog.dismiss()
+            }
+            .create()
+    }
+
+    /*
+     * Function will create a QR code for the Tox ID and assign it to the image view if specified.
+     * Then it will return the QR code bitmap with the specified padding.
+     */
+    private fun createQrCode(
+        qrCodeColor: Int = Color.BLACK,
+        backgroundColor: Int = Color.WHITE,
+        qrSize: Size = Px(
+            min(
+                resources.displayMetrics.widthPixels,
+                resources.displayMetrics.heightPixels
+            ) * QR_CODE_TO_SCREEN_RATIO
+        ),
+        padding: Size = qrCodePadding,
+        imageView: ImageView? = null,
```

Comment:
This function probably shouldn't do anything with an imageview. Why not just return the QR code and set it into the imageview at the call point?